client/scripts/views/pages/view_snapshot_proposal/index.ts

```diff
@@ -260,41 +305,152 @@ const ViewProposalPage: m.Component<{
+                      m('', 'Author'),
+                      m('', 'IPFS'),
+                      m('', 'Voting System'),
+                      m('', 'Start Date'),
+                      m('', 'End Date'),
+                      m('', 'Snapshot'),
+                    ]),
+                    m('.values', [
+                      m(User, {
+                        user: new AddressInfo(
+                          null,
+                          proposal.author,
+                          app.activeId(),
+                          null
+                        ),
+                        linkify: true,
+                        popover: true,
+                      }),
+                      m(
+                        'a',
+                        {
+                          class: 'snapshot-link -mt-10',
+                          href: `https://ipfs.fleek.co/ipfs/${proposal.ipfs}`,
+                          target: '_blank',
+                        },
+                        [
+                          m('.truncate', `#${proposal.ipfs}`),
+                          m(Icon, { name: Icons.EXTERNAL_LINK, class: 'ml-1' }),
+                        ]
+                      ),
+                      m(
+                        '.snapshot-type',
+                        proposal.type.split('-').join(' ').concat(' voting')
+                      ),
+                      m('', moment(+proposal.start * 1000).format('lll')),
+                      m('', moment(+proposal.end * 1000).format('lll')),
+                      m(
+                        'a',
+                        {
+                          class: 'snapshot-link',
+                          href: `https://etherscan.io/block/${proposal.snapshot}`,
+                          target: '_blank',
+                        },
+                        [
+                          m('.truncate', `#${proposal.snapshot}`),
+                          m(Icon, { name: Icons.EXTERNAL_LINK, class: 'ml-1' }),
+                        ]
+                      ),
+                    ]),
+                  ]),
+                  thread !== 'false' &&
+                    m('.padding-x-20.linked-discussion', [
+                      m('.heading-2', 'Linked Discussion'),
+                      m(ProposalHeaderSnapshotThreadLink, {
+                        threadId: vnode.state.thread,
+                      }),
+                    ]),
+                ]),
+                isActive &&
+                  author &&
+                  m(VoteAction, {
+                    space: vnode.state.space,
+                    proposal: vnode.state.proposal,
+                    id: vnode.attrs.identifier,
+                    totalScore: vnode.state.totalScore,
+                    scores: vnode.state.scores,
+                    choices: vnode.state.proposal.choices,
+                    votes: vnode.state.votes,
+                  }),
+                m('.proposal-info-box.padding-x-20.mt-30', [
```

Comment:
factor out things like this into the scss mixin, and put in a human readable class-naming convention here